### PR TITLE
tests for retainLines formatting

### PR DESCRIPTION
**Pull request by kevinb7**
Adds tests for https://github.com/babel/babel/commit/16f7b967b5f7bcb6572c2a47de70ee600702e31e.

### CODE REVIEW

File: test/core/formatting.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        assert.equal(out, '"use strict";\n\nvar div = React.createElement(\n  "div",\n  null,\n  "test"\n);');
+    });
+
+    it("should not pretty print JSXElements with children when retainLines = true", function() {
+        var src = "var div = <div>test</div>";
+        var out = transform(src, { retainLines: true }).code;
+
+        assert.equal(out, '"use strict";var div = React.createElement("div", null, "test");');
+    });
+});
```
Comment (kevinb7) [inline test/core/formatting.js]: Let me know if you'd like fixtures for this instead.

**Comment by sebmck** [comment]
Would be best to have them in `fixtures` since it's kinda gross to have them separate when you can just put them alongside the others.

**Comment by kevinb7** [comment]
Should I create a ```fixtures/formatting``` folder?  If I put the fixtures in one of the other existing subfolders I'm not sure how to control the options on a per fixture basis.

**Comment by sebmck** [comment]
These are generation tests so they'd belong in [test/core/fixtures/generation](https://github.com/babel/babel/tree/master/test/core/fixtures/generation). Likely in the `edgecase` folder. You can see an example test fixture that uses the `retainLines` option [here](https://github.com/babel/babel/tree/master/test/core/fixtures/generation/edgecase/return-with-retainlines-option).

**Comment by kevinb7** [comment]
Awesome, thanks!

**Comment by kevinb7** [comment]
I'm going to add blank lines to the ends of the fixtures.

**Comment by kevinb7** [comment]
Should be good to go now.

**Comment by sebmck** [comment]
Why's the folder prepended with a dot? It's going to be ignored by the test runner.

**Comment by kevinb7** [comment]
I was looking for example tests that used the "react" transform so that the expected would contain ```React.createElement(...)``` instead of ```<div>```.  They happened to have to have a ```.``` in front of the folder and assumed that's signified that's what specified that "react" transform should be run.  I also tried adding ```whitelist: ["react"]```.

**Comment by sebmck** [comment]
No transforms are ran for the generation tests.

**Comment by sebmck** [comment]
Although that completely negates the purpose of the test so I guess `transformation` is the right place after all. Sorry for the confusion.

**Comment by kevinb7** [comment]
Thanks for your help with these tests.  Also, thanks for all of your hard work on babel!

**Comment by kevinb7** [comment]
This time I actually verify that indeed the test was running.

**Comment by sebmck** [comment]
Rad, thank you @kevinb7!

**Pull request merged**